### ISSUE TITLE
Exclude target directory

### ISSUE THREAD

**Issue by lbeuster** [marco-nicolini/shellcheck-maven-plugin#1]
I want to set the sourceDirectory to the root of the project. Unfortunately the direct target dir and also the target dirs of sub-modules are checked.
Maybe it makes sense to add excludes as config params. Or as an alternative or additionally to add a flag to disable traversal of sub-dirs of the sourceDirectories.

**Comment by lbeuster** [comment]
Or maybe more generic
```
<sourceItems>
  <sourceItem>
    <dir>...</dir>
    <traverseSubDirs>true/false</traverseSubDirs>
    <inludes>
      <include>ant-style-pattern</include>
    </includes>
    <excludes>
      <exclude>ant-style-pattern</exclude>
    </excludes>
  <sourceItem>
</sourceItems>
```

**Comment by giraciopide** [comment]
Makes completely sense.
I'll change how source dirs are specified to use something that extends FileSet (from the maven-model). I believe this should give you the standard expected include/exclude mechanism with ant-like patterns.

**Comment by giraciopide** [comment]
I believe your use case is now possible with this configuration:

```xml
  <sourceDirs>
      <sourceDir>
          <directory>${project.basedir}</directory>
          <includes>
              <include>**/*.sh</include>
          </includes>
          <excludes>
              <exclude>**/target/**</exclude>
          </excludes>
      </sourceDir>
  </sourceDirs>
```
Next days I'll release and push to maven central. Meanwhile if you need/want you can build the dev branch.

**Comment by lbeuster** [comment]
Yes, thanks, this would be great ;)

**Comment by giraciopide** [comment]
I released on central version 0.3.0 that contains this.
It will be available in some hours (time to sync from the sonatype repo to actual maven central).
Let me know if you have problems and I'll look into them.

**Comment by lbeuster** [comment]
Thanks, works fine. A nice default config (for pluginManagement) could be this:
```
          <configuration>
            <sourceDirs>
              <sourceDir>
                <directory>.</directory>
                <includes>
                  <include>**/*.sh</include>
                </includes>
                <excludes>
                  <exclude>**/target/**</exclude>
                </excludes>
              </sourceDir>
            </sourceDirs>
          </configuration>
```
